tikv-client/src/main/java/com/pingcap/tikv/operation/SchemaInfer.java

```diff
@@ -98,7 +98,7 @@ private void buildTransform(TiDAGRequest dagRequest) {
   private void extractFieldTypes(TiDAGRequest dagRequest) {
 
     if (dagRequest.hasAggregate()) {
-      dagRequest.getAggregates().forEach(expr -> types.add(expr.getType()));
+      dagRequest.getAggregatePairs().forEach(pair -> types.add(pair.second));
```

Comment:
There are some strange things involving logic here.  As a matter of fact `sum(col)` is not really returning the same type as `col`.

e.g., 
`sum(tp_int)` returns NEW_DECIMAL flag, 
`sum(tp_float)` returns DOUBLE flag,
`sum(tp_double)` returns DOUBLE flag,
`sum(tp_decimal)` returns NEW_DECIMAL flag.

plz confirm whether logic here is correct with some tests.

Reply:
Working on this.

Reply:
Based on the private discussion with engineer from TiDB, `sum` logic is now compatible with `MySQL`. That is being said `sum` will return as same type as `column`. Are testing against with the commit hashes provided above?

Reply:
You may add `--column-info-type` flag at tidb startup and take a look at the detailed type information.

As a brief example, check this: `mysql -h 127.0.0.1 -P 4000 -u root --column-info-type`

Reply:
Above return flags are what I observed on tidb's master branch and raw mysql.

Reply:
Also it should be quite obvious that statements like `sum(int)` should not return `int` type since it is too easy to overflow.

Reply:
You are absolutely right. I misunderstood sum logic when I communicated with Engineer from TiDB. To me, it is ood too if sum returns as same type as column. They only change return type logic for double. More precisely, if column is double then sum return double. If column is int or decimal, then sum still returns decimal. That is being said we need know what type that a column carry at run time.  Expression does not carry any column information right now. In addtion, type system is currently refactoring. Hence this PR will be blocked until type system is down with refractory. Lastly, we need figure out a why to do version. If we merge this change, then tispsrk is not compatible with any version before or at TiDB 1.05.